Automattic/sensei
# Extract WPML API and create duplicates when they are needed

In this PR I address an issue that was found previously: https://github.com/Automattic/sensei/pull/7492#issuecomment-2021387041

Unfortunately, questions are not translated automatically. (You can see these strings in the WPML translation editor, but they don't appear in the translated lesson.)
For now, the user has to translate them manually in lessons: open the translated lesson in the WordPress editor and change texts for questions.

## Proposed Changes
* Extract WPML API. Implemented as a trait for simplicity.
* Create translations (duplicates) for lessons if the needed translation direction doesn't exist yet.
* Create or update translations (duplicates) for quizzes and questions. 

## Testing Instructions

1. Install and activate WPML.
2. Set up at least 2 additional languages.
3. Create an original course with lessons and quizzes.
4. Translate it to the first "additional" language.
5. Make sure you have translated lessons and questions in the target language.
6. Translate it to the second "additional" language.
7. Again, make sure you have translated lessons and questions in the target language. 

## Pre-Merge Checklist
